library-conan/CMakeLists.txt

```diff
@@ -0,0 +1,315 @@
+message("-- INFO: Conan detected... finding packages")
+find_package(OpenSSL REQUIRED COMPONENTS SSL Crypto)
+find_package(ZLIB REQUIRED)
+find_package(libsodium REQUIRED)
+find_package(Opus)
+find_package(Threads REQUIRED)
+find_package(Git QUIET)
+
```

Comment:
I understand that using Conan might need some ``find_package()`` that maybe are not used when not using Conan because some other strategy like ``FetchContent`` is done, but why the big CMakeLists.txt and all the code below? In theory Conan can integrate quite cleanly, without requiring modifying CMakeLists code.

Reply:
We wanted the CMakeLists.txt to be separate (see library-vcpkg), this is just something we do with VCPKG so we wanted the same format to follow.

Reply:
It should also be noted, our main CMakeLists.txt doesn't do everything (hence why we have `library/CMakeLists.txt`, `library-vcpkg/CMakeLists.txt`, and now `library-conan/CMakeLists.txt`)